pkg/edit/navigation_test.go

```diff
@@ -63,23 +66,91 @@ func TestNavigation_WidthRatio(t *testing.T) {
 	evals(f.Evaler, `@edit:navigation:width-ratio = 1 1 1`)
 	f.TTYCtrl.Inject(term.K('N', ui.Ctrl))
 	f.TestTTY(t,
-		"~"+string(os.PathSeparator)+"d> ", term.DotHere, "\n",
+		filepath.Join("~", "d"), "> ", term.DotHere, "\n",
 		" NAVIGATING  \n", Styles,
 		"************ ",
-		" d                a               ", Styles,
+		" d                a               \n", Styles,
 		"################ ++++++++++++++++ ",
+		"                  e              ", Styles,
+		"                 ////////////////",
+	)
+}
+
+// Test some corner cases:
+//
+// 1) Inserting a selection when the CLI cursor is at the start of the edit
+// buffer omits the space char prefix.
+//
+// 2) Inserting the "selection" in an empty directory inserts nothing rather
+// than the literal ` ''`  that used to be inserted.
+//
+// 3) Inserting a selection when the CLI cursor is not at the start of the
+// edit buffer but the preceding char is a space does not prepend another
+// space.
+//
+// See https://github.com/elves/elvish/issues/1169
+func TestNavigationCornerCases(t *testing.T) {
```

Comment:
Please break this test into multiple tests.

Reply:
Done